### ISSUE TITLE
 Check failed happened when use euclidean layer on VGG_FACE 

### ISSUE THREAD

**Issue by 139419qw** [BVLC/caffe#6132]
Below is my train.prototxt:
```
layer {
  name: "data"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TRAIN
  }
  transform_param {
    mirror: true
    crop_size: 224
    mean_file: "/home/rnd/caffe-master/data/faceRec/data/faces_mean.binaryproto"
  }
  data_param {
    source: "/home/rnd/caffe-master/data/faceRec/lmdb/faces_train_lmdb"
    batch_size: 15
    backend: LMDB
  }
}
layer {
  name: "data"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TEST
  }
  transform_param {
    mirror: false
    crop_size: 224
    mean_file: "/home/rnd/caffe-master/data/faceRec/data/faces_mean.binaryproto"
  }
  data_param {
    source: "/home/rnd/caffe-master/data/faceRec/lmdb/faces_val_lmdb"
    batch_size: 6
    backend: LMDB
  }
}
...
...
...
layer {
  name: "drop7"
  type: "Dropout"
  bottom: "fc7"
  top: "fc7"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc9"
  type: "InnerProduct"
  bottom: "fc7"
  top: "fc9"
  param {
    lr_mult: 2
    decay_mult: 0
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 3
    weight_filler {
      type: "xavier"
    }
  }
}
layer {
  name: "accuracy_1"
  type:  "Accuracy"
  bottom: "fc9"
  bottom: "label"
  top: "accuracy"
  include: { phase: TEST }
}
layer {
  name: "loss_cls"
  type: "EuclideanLoss"
  bottom: "fc9"
  bottom: "label"
  top: "loss_cls"
  loss_weight: 1
}
```
And I get this error:
```
F1225 17:12:29.952535  6819 euclidean_loss_layer.cpp:12] Check failed: bottom[0]->count(1) == bottom[1]->count(1) (5 vs. 1) Inputs must have the same dimension.
```
I only have 3 persons to classify and the output of fc9 is 3 too, Why I still get this problem?When I changed "EuclideanLoss" with "SoftmaxWithLoss", this net can work well.
Can anyone tell me why?Please!

**Comment by Noiredd** [comment]
Consult the documentation on [EuclideanLossLayer](http://caffe.berkeleyvision.org/doxygen/classcaffe_1_1EuclideanLossLayer.html#details) and [SoftmaxWithLossLayer](http://caffe.berkeleyvision.org/doxygen/classcaffe_1_1SoftmaxWithLossLayer.html#details). Tl;DR: Euclidean calculates sum of squares of differences between elements of its bottoms (thus requiring them to be of equal size), SoftmaxWithLoss calculates a multinomial loss function versus a label which is just an index of the correct class (thus for each example in the batch you only need 1 element).
